source/index.ts

```diff
@@ -0,0 +1,746 @@
+		return semver.clean(version) === null;
+	}
+
+	_shouldPerformMigration(candidateVersion: string, previousMigratedVersion: string, versionToMigrate: string): boolean {
+		if (this._isVersionInRangeFormat(candidateVersion)) {
+			if (previousMigratedVersion !== '0.0.0' && semver.satisfies(previousMigratedVersion, candidateVersion)) {
+				return false;
+			}
+
+			return semver.satisfies(versionToMigrate, candidateVersion);
+		}
+
+		if (semver.lte(candidateVersion, previousMigratedVersion)) {
+			return false;
+		}
+
+		if (semver.gt(candidateVersion, versionToMigrate)) {
+			return false;
+		}
+
+		return true;
+	}
+
+	_get(key: string, defaultValue?: unknown): any {
+		return dotProp.get(this.store, key, defaultValue);
+	}
+
+	_set(key: string, value?: unknown): void {
+		const {store} = this;
+		dotProp.set(store, key, value);
+
+		this.store = store;
+	}
+
+	/**
+	Get an item.
+
+	@param key - The key of the item to get.
+	@param defaultValue - The default value if the item does not exist.
+	*/
+	get(key: string, defaultValue?: unknown): unknown {
+		if (this._options.accessPropertiesByDotNotation) {
+			return dotProp.get(this.store, key, defaultValue);
+		}
+
+		return key in this.store ? this.store[key] : defaultValue;
+	}
+
+	/**
+	Set an item.
+
+	@param key - You can use [dot-notation](https://github.com/sindresorhus/dot-prop) in a key to access nested properties.
+	@param value - Must be JSON serializable. Trying to set the type `undefined`, `function`, or `symbol` will result in a `TypeError`.
+	*/
+	set(key: string | {[key: string]: unknown}, value?: unknown): void {
+		if (typeof key !== 'string' && typeof key !== 'object') {
+			throw new TypeError(`Expected \`key\` to be of type \`string\` or \`object\`, got ${typeof key}`);
+		}
+
+		if (typeof key !== 'object' && value === undefined) {
+			throw new TypeError('Use `delete()` to clear values');
+		}
+
+		if (this._containsReservedKey(key)) {
+			throw new TypeError(`Please don't use the ${INTERNAL_KEY} key, as it's used to manage this module internal operations.`);
+		}
+
+		const {store} = this;
+
+		const set = (key: string, value: unknown): void => {
+			checkValueType(key, value);
+			if (this._options.accessPropertiesByDotNotation) {
+				dotProp.set(store, key, value);
+			} else {
+				store[key] = value;
+			}
+		};
+
+		if (typeof key === 'object') {
+			const object = key;
+			for (const [key, value] of Object.entries(object)) {
+				set(key, value);
+			}
+		} else {
+			set(key, value);
+		}
+
+		this.store = store;
+	}
+
+	/**
+	Check if an item exists.
+
+	@param key - The key of the item to check.
+	*/
+	has(key: string): boolean {
+		if (this._options.accessPropertiesByDotNotation) {
+			return dotProp.has(this.store, key);
+		}
+
+		return key in this.store;
+	}
+
+	/**
+	Reset items to their default values, as defined by the `defaults` or `schema` option.
+
+	@param keys - The keys of the items to reset.
+	*/
+	reset(...keys: string[]): void {
```

Comment:
Same as above.

Reply:
What typing do you suggest here? Because `string` is already equal to `K` (which is just the key of Conf)

Reply:
The answer is in the existing TS definition:

https://github.com/sindresorhus/conf/blob/97bcc918a109ba964f7ef8a091cc5a762ae03576/index.d.ts#L289

Yes, both resolve to a string, but having it strongly typed will let editors auto-complete the key names and let TS enforce only valid key names.

Reply:
I've not been able to make it work, it always says this:

![image](https://user-images.githubusercontent.com/24696635/74832561-7c1b4c00-530f-11ea-99e6-2d6e64e691cf.png)

Do you have any suggestions for my problem? :/